### PR TITLE
Refactor to allow merge

### PR DESCRIPTION
**Pull request by chasscepts**
We implemented the following specifications of the project
- A default project 'Main' is added if storage is empty.
- A user can add new Todo.
- A User can create a new project by providing an unexisting project when creating a Todo.
- A user can view the list of all Todos grouped under a project.
- A user can view the details, edit and delete a Todo.
- A user can delete a project and all the Todos in it.
- Deleting a project requires confirmation to avoid accidental deletion.

### CODE REVIEW

File: src/js/form.js [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  if (selectedProject && selectedProject !== li) {
+    selectedProject.classList.remove('selected');
+  }
+  selectedProject = li;
+  selectedProject.classList.add('selected');
+  todosContainer.innerHTML = '';
+  db.getAll(li.textContent).forEach((todo) => addToDo(todo));
+};
+
+export default function ToDoForm() {
```
Comment (rindrajosia) [inline src/js/form.js]: Please use arrow function

File: src/js/todo.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export default function Todo(title, description, duedate, priority, isDone) {
```
Comment (rindrajosia) [inline src/js/todo.js]: Please use arrow function

File: src/js/form.js [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+  const overlay = document.querySelector('.overlay');
+  const form = document.querySelector('.form');
+  const titleInput = form.querySelector('#title');
+  const descriptionInput = form.querySelector('#description');
+  const dateInput = form.querySelector('#date');
+  const projectInput = form.querySelector('#proj');
+  const priorityInput = form.querySelector('#priority');
+
+  const show = (callback, todo) => {
+    overlay.style.display = 'block';
```
Comment (rindrajosia) [inline src/js/form.js]: Keep all the styling logic in CSS files

File: src/js/form.js [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+      titleInput.value = '';
+      descriptionInput.value = '';
+      priorityInput.value = '';
+      dateInput.value = '';
+      projectInput.value = '';
+    }
+
+    form.onsubmit = (evt) => {
+      evt.preventDefault();
+      overlay.style.display = 'none';
```
Comment (rindrajosia) [inline src/js/form.js]: Keep all the styling logic in CSS files

File: src/js/form.js [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        titleInput.value, descriptionInput.value, dateInput.value, priorityInput.value, false,
+      );
+      callback({ todo, project });
+    };
+  };
+
+  return {
+    show,
+    close: () => {
+      overlay.style.display = 'none';
```
Comment (rindrajosia) [inline src/js/form.js]: Keep all the styling logic in CSS files

File: src/js/form.js [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+};
+
+const addToDo = (todo) => {
+  const card = document.createElement('div');
+  card.classList.add('card');
+  card.classList.add(todo.priority);
+
+  const title = document.createElement('h3');
+  title.textContent = todo.title;
+  title.style.display = 'inline';
```
Comment (rindrajosia) [inline src/js/form.js]: Keep all the styling logic in CSS files

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- It uses webpack :+1: 
- There is a 'default' project :100: 
- It uses localStorage to save todos between sessions :100: 
- User can view all projects :+1: 

To improve :factory_worker: [REQUIRED]:

- [x] Please make sure that the information in your readme is correct because it is the first thing the user checks. I am not able to add inline comments on your readme because there is just one commit in this branch but, check the project description because you talked about Library/books in this section in addition, in the Getting Started section, npm run dev doesn't work. It should be npm run serve
- [x] It will be nice if you use arrow function instead of function keyword because you are ES6 syntax
- [x] The user should be able to create new projects ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/projects_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/todo-list))
- [x] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright)
- [x] Please commit more

Nice to have:

- [ ] 

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by mahmoud717 — COMMENTED** [review]
### STATUS: APPROVED ✔️ ✔️ 

**hi @ashraffares  &  @chasscepts**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**

**Pull request merged**